src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1382,21 +1352,63 @@ void walkInnerSymbols(ObjectType type, String innerNamespace) {
           privateEnums.add(innerNamespace + '.' + propName);
         }
       }
+      boolean nonEmpty = false;
 
       for (String propName : getSortedPublicPropertyNames(type)) {
         String qualifiedName = innerNamespace + '.' + propName;
         if (provides.contains(qualifiedName)) continue;
         JSType pType = type.getPropertyType(propName);
-
         if (pType.isEnumType()) {
+          if (!nonEmpty) {
+            emitNamespaceBegin(innerNamespace);
```

Comment:
That pattern got us in trouble before, because "NamespaceMembers" is tricky to define (4 branches on type - class / interface / function / enum, then 2 branches on member - function / field). So we had a mismatch where we didn't catch a namespaceMember, but would have emitted it. Also I am still not confident I got it right (closure API is hard - see isNominalConstructor that returns true on structural interfaces ?!?), so rather keep it in one place.

Going with the rename and single lining the emit, for better readability.